implementation/src/main/java/io/smallrye/openapi/runtime/scanner/dataobject/WrappedIndexView.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2018 Red Hat, Inc, and individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.smallrye.openapi.runtime.scanner.dataobject;
+
+import io.smallrye.openapi.runtime.util.TypeUtil;
+import org.jboss.jandex.AnnotationInstance;
+import org.jboss.jandex.ClassInfo;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.IndexView;
+import org.jboss.jandex.Type;
+
+import javax.validation.constraints.NotNull;
+import java.util.Collection;
+
+/**
+ * @author Marc Savy {@literal <marc@rhymewithgravy.com>}
+ */
+public class WrappedIndexView implements IndexView {
```

Comment:
Perhaps a better name for this class?  Yes it's wrapping an `IndexView` - but **why** is it wrapping it?  I have to open it up and look to know that.  Maybe `EnhancedIndexView`?  That's better though probably not much. :)